meson-scripts/test-cc-supports-hidden-visibility.py

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import sys
+import subprocess
+
+# Put output files in the builddir
+os.chdir(os.environ['MESON_BUILD_ROOT'])
+
+infile = 'visibility-conftest.c'
+outfile = 'visibility-conftest.S'
+
+with open(infile, 'w') as f:
+    f.write('int __attribute__ ((visibility ("hidden"))) foo (void) { return 1; }')
+
+args = sys.argv[1:]
+args += ['-Werror', '-S', infile, '-o', outfile]
+res = subprocess.run(args, stdout=subprocess.PIPE, universal_newlines=True, check=True)
+
+with open(outfile, 'r') as f:
+    if re.search('\.hidden.*foo', f.read()):
+        sys.exit(0)
+print('.hidden not found in the outputted assembly')
```

Comment:
That test comes from @nirbheek meson port: https://gitlab.freedesktop.org/gstreamer/meson-ports/libffi/-/blob/meson/test-cc-supports-hidden-visibility.py and is is a direct translation of what configure.ac has.

I don't know if it's intentional or not, but that test does not actually compile the code, it translates to assembly and checks if the asm code has `.hidden` instruction which libffi asm uses.

I'm pretty sure this can be replaced by `cc.has_function_attribute('visibility:hidden')`, what do you think?

Reply:
Actually, @nirbheek code probably predates the `visibility:hidden` syntax, that could explain it.